calico_rkt/calico_rkt.py

```diff
@@ -1,91 +1,249 @@
-        "netgroups": ["prod", "frontend"], # Required.
-        "labels": {  # Optional.
-            "rack": "3A",
-            "pop": "houston"
-    }
-    """
-    pass
+        print_stderr("This container has already been configured with Calico Networking.")
+        sys.exit(1)
 
+    # Allocate and Assign ip address through IPAM Client
+    ip = _allocate_IP()
 
-def isolate(args):
+    # Create Endpoint object
+    try:
+        ep = client.create_endpoint(HOSTNAME, ORCHESTRATOR_ID,
+                                      container_id, [IPAddress(ip)])
+    except AddrFormatError:
+        print_stderr("This node is not configured for IPv%d. Unassigning IP "\
+                      "address %s then exiting."  % ip.version, ip)
+        client.unassign_address(IPNetwork(input_json['ipam']['subnet']), ip)
+        sys.exit(1)
+
+    # Create the veth, move into the container namespace, add the IP and
+    # set up the default routes.
+    ep.mac = ep.provision_veth(Namespace(netns_path), interface)
+    client.set_endpoint(ep)
+
+    # Let the caller know what endpoint was created.
+    return ep, ip
+
+def _container_remove(hostname, orchestrator_id, container_id, client):
     """
-    "args": {
-        "hostname": "slave-H3A-1", # Required
-        "container-id": "ba11f1de-fc4d-46fd-9f15-424f4ef05a3a", # Required
-        "pid": 3789 # Required
-    }
+    Remove the indicated container on this host from Calico networking
     """
-    pass
+    # Find the endpoint ID. We need this to find any ACL rules
+    try:
+        endpoint = client.get_endpoint(
+            hostname=hostname,
+            orchestrator_id=orchestrator_id,
+            workload_id=container_id
+        )
+    except KeyError:
+        print_stderr("Container %s doesn't contain any endpoints" % container_id)
+        sys.exit(1)
 
+    # Remove any IP address assignments that this endpoint has
+    for net in endpoint.ipv4_nets | endpoint.ipv6_nets:
+        assert(net.size == 1)
+        client.unassign_address(None, net.ip)
 
-def update(args):
-    pass
+    # Remove the endpoint
+    netns.remove_veth(endpoint.name)
 
+    # Remove the container from the datastore.
+    client.remove_workload(hostname, orchestrator_id, container_id)
 
-def cleanup(args):
+    print_stderr("Removed Calico interface from %s" % container_id)
+
+def _create_profile(endpoint, profile_name, ip, client):
     """
-    "args": {
-        "hostname": "slave-H3A-1", # Required
-        "container-id": "ba11f1de-fc4d-46fd-9f15-424f4ef05a3a" # Required
-    }
+    Configure the calico profile for a pod.
+
+    Currently assumes one pod with each name.
     """
-    pass
+    print_stderr('Configuring Pod Profile: %s' % profile_name)
 
+    if client.profile_exists(profile_name):
+        print_stderr("Profile with name %s already exists, applying to endpoint." % (profile_name))
 
+    else:
+        client.create_profile(profile_name)
```

Comment:
With the printing you have, it would make sense to have a line saying the profile is being created here.